### PR TITLE
feat: add support for codepipeline test reports

### LINKED ISSUE #2289: Support for CodeBuild Test Reports
In order to support the CodeBuild Test Reports feature which allows viewing test results in the AWS Console it appears that the `templates/cicdf/pipeline_cfn.yml` file needs to be modified expanding the `BuildProjectPolicy` to include the following statement.

```
- Effect: Allow
            Action:
              - codebuild:CreateReportGroup
              - codebuild:CreateReport
              - codebuild:UpdateReport
              - codebuild:BatchPutTestCases
              - codebuild:BatchPutCodeCoverages
            Resource: !Sub arn:aws:codebuild:${AWS::Region}:${AWS::AccountId}:report-group/pipeline-{{$.AppName}}-*

```

I have made these changes on my own fork and have successfully been able to view test reports in CodeBuild as a result.  At this point I'm happy to submit a PR however I believe I need to update the end-to-end tests in order for the PR to be accepted.  From what I can tell this may just require adding the necessary cf to the `internal/pkg/deploy/cloudformation/stack/testdata/pipeline/bb_template.yml`?  

I'm new to golang and this project so any direction would be greatly appreciated.

### PR DESCRIPTION
**Pull request by patnolanireland**
Resolves #2289

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -141,6 +141,7 @@ github.com/gobuffalo/logger v1.0.3 h1:YaXOTHNPCvkqqA7w05A4v0k2tCdpr+sgFlgINbQ6gq
 github.com/gobuffalo/logger v1.0.3/go.mod h1:SoeejUwldiS7ZsyCBphOGURmWdwUFXs0J7TCjEhjKxM=
 github.com/gobuffalo/packd v1.0.0 h1:6ERZvJHfe24rfFmA9OaoKBdC7+c9sydrytMg8SdFGBM=
 github.com/gobuffalo/packd v1.0.0/go.mod h1:6VTc4htmJRFB7u1m/4LeMTWjFoYrUiBkU9Fdec9hrhI=
+github.com/gobuffalo/packr v1.30.1 h1:hu1fuVR3fXEZR7rXNW3h8rqSML8EVAf6KNm0NKO/wKg=
```
Comment (Lou1415926) [inline go.sum:144]: Could you run `go mod tidy` to see if this change remains?

**Review by Lou1415926 — COMMENTED** [review]
Thank you so much again for your contribution 🙇🏼‍♀️  The update to template looks good to me!

Could you remove the `tags` file since it seems to be accidentally committed? Thanks!!

Edit: 
To pass the "Semantic Pull Request" check, you can simply add one more commit, or update your current commit message with a prefix of "chore:". 

"Semantic Pull Request" requires that you either have multiple commits in a PR, or have a single commit that meets the semantic pull request requirements, which is having a prefix such as "chore", "docs", "fix", etc.

**Comment by patnolanireland** [comment]
Apologies I didn't realise I'd submitted this PR.  It was off an old branch, the new PR is up as https://github.com/aws/copilot-cli/pull/2306.

File: go.sum (hunk 1 above)
Comment (patnolanireland) [reply go.sum:144]: I've run this against the branch the new PR is based off.  The change remained, i.e. nothing to `git commit`.

**Pull request closed without merging**